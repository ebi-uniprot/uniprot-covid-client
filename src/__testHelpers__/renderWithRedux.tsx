/* eslint-disable */
import React from 'react';
import { createStore, applyMiddleware } from 'redux';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { createMemoryHistory, MemoryHistory } from 'history';
import rootReducer from '../state/rootReducer';
import { Router, Route } from 'react-router-dom';

type RenderOptions = {
  route?: string;
  history?: MemoryHistory<any>;
  path?: string;
  initialState?: any;
  store?: any;
};

const renderWithRedux = (
  ui: React.Component,
  {
    route = '',
    history = createMemoryHistory({ initialEntries: [route] }),
    path,
    initialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
  }: RenderOptions = {}
) => {
  return {
    ...render(
      <Provider store={store}>
        <Router history={history}>
          {path ? <Route path={path} render={() => ui} /> : ui}
        </Router>
      </Provider>
    ),
    store,
    history,
  };
};

export default renderWithRedux;
