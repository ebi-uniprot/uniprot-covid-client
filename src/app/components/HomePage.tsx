import React, { lazy, Suspense } from 'react';
import { HeroHeader, Loader } from 'franklin-sites';

import SearchContainer from '../../uniprotkb/components/search/SearchContainer';
import ErrorBoundary from '../../shared/components/error-component/ErrorBoundary';

const HomePageNonCritical = lazy(() =>
  import(
    /* webpackChunkName: "home-page-non-critical" */ './HomePageNonCritical'
  )
);
const UniProtFooter = lazy(() =>
  import(
    /* webpackChunkName: "footer" */ '../../shared/components/layouts/UniProtFooter'
  )
);

const mission =
  'The mission of UniProt is to provide the scientific community with a comprehensive, high-quality and freely accessible resource of protein sequence and functional information.';

const HomePage = () => (
  <>
    <main>
      <ErrorBoundary>
        <HeroHeader title="Find your protein" footer={mission}>
          <SearchContainer />
        </HeroHeader>
      </ErrorBoundary>

      <ErrorBoundary>
        <Suspense fallback={<Loader />}>
          <HomePageNonCritical />
        </Suspense>
      </ErrorBoundary>
    </main>
    <Suspense fallback={null}>
      <ErrorBoundary>
        <UniProtFooter />
      </ErrorBoundary>
    </Suspense>
  </>
);

export default HomePage;
