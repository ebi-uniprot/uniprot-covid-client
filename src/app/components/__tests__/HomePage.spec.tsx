import React from 'react';

import renderWithRedux from '../../../shared/__test-helpers__/RenderWithRedux';

import HomePage from '../HomePage';

describe('HomePage component', () => {
  test('should render', () => {
    const { asFragment } = renderWithRedux(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });
});
