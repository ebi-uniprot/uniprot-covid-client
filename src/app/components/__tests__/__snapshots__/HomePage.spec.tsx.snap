// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should render 1`] = `
<DocumentFragment>
  <main>
    <div
      class="hero-header"
    >
      <div
        class="hero-header__title"
      >
        <h1>
          Find your protein
        </h1>
      </div>
      <div
        class="hero-header__content"
      >
        <form
          class="main-search"
          data-testid="main-search-form"
        >
          <div
            class="dropdown-container"
          >
            <button
              class="button dropdown "
              type="button"
            >
              UniProtKB
            </button>
            <div
              class="dropdown-menu"
            >
              <ul>
                <li>
                  <button
                    type="button"
                  >
                    UniProtKB - the UniProt knowledgebase
                  </button>
                </li>
                <li>
                  <button
                    type="button"
                  >
                    UniRef
                  </button>
                </li>
                <li>
                  <button
                    type="button"
                  >
                    UniParc
                  </button>
                </li>
                <li>
                  <button
                    type="button"
                  >
                    Proteomes
                  </button>
                </li>
                <li>
                  <button
                    type="button"
                  >
                    Publications
                  </button>
                </li>
                <li>
                  <button
                    type="button"
                  >
                    Keywords
                  </button>
                </li>
              </ul>
            </div>
          </div>
          <input
            class="main-search__input"
            data-testid="main-search-input"
            type="text"
            value=""
          />
          <button
            class="button"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="hero-header__footer"
      >
        The mission of UniProt is to provide the scientific community with a comprehensive, high-quality and freely accessible resource of protein sequence and functional information.
      </div>
    </div>
    <div
      class="loader-container"
    >
      <svg
        class="loader"
        height="100"
        viewBox="0 0 100 100"
        width="100"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M9 50c6.833-10.667 13.667-16 20.5-16S43.167 39.333 50 50s13.667 16 20.5 16S84.167 60.667 91 50"
            stroke="#3ca3ca"
            stroke-linecap="square"
            stroke-width="2"
          />
          <circle
            cx="9"
            cy="50"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="30"
            cy="34"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="50"
            cy="50"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="70"
            cy="66"
            fill="#3ca3ca"
            r="6"
          />
          <circle
            cx="91"
            cy="50"
            fill="#3ca3ca"
            r="6"
          />
        </g>
      </svg>
    </div>
  </main>
</DocumentFragment>
`;
